ethportal-peertest/src/jsonrpc.rs

```diff
@@ -144,22 +189,22 @@ fn get_response_result(response: Value) -> Result<Value, JsonRpcResponseError> {
     }
 }
 
-pub async fn test_jsonrpc_endpoints_over_http(target_http_address: String) {
+pub async fn test_jsonrpc_endpoints_over_http(peertest_config: PeertestConfig) {
     let client = Client::new();
-    for endpoint in JsonRpcEndpoint::all_endpoints() {
+    for endpoint in JsonRpcEndpoint::all_endpoints(peertest_config.target_node) {
         info!("Testing over HTTP: {:?}", endpoint.method);
-        let json_string = endpoint.to_jsonrpc();
+        let json_string = &endpoint.to_jsonrpc();
         let req = Request::builder()
             .method(Method::POST)
-            .uri(format!("http://{}", target_http_address))
+            .uri(format!("http://{}", peertest_config.target_http_address))
             .header("content-type", "application/json")
-            .body(Body::from(json_string))
+            .body(Body::from(json_string.clone()))
```

Comment:
Why the borrow in line 196, followed by a clone here? I don't see why either line needs to change at all.